spec/i18n_spec.rb

```diff
@@ -258,24 +256,56 @@ def allowed_untranslated_key?(locale, key)
     missing_interpolation_argument_locale_keys = []
 
     i18n.data[i18n.base_locale].select_keys do |key, _node|
-      if key.start_with?('i18n.transliterate.rule.') || i18n.t(key).is_a?(Array) || i18n.t(key).nil?
+      if key.start_with?('i18n.transliterate.rule.') || i18n.t(key).is_a?(Array) || !i18n.t(key)
         next
       end
 
       interpolation_arguments = i18n.locales.map do |locale|
-        if ALLOWED_INTERPOLATION_MISMATCH_LOCALE_KEYS.include?("#{locale}.#{key}")
-          missing_interpolation_argument_locale_keys.push("#{locale}.#{key}")
-          next
+        value = extract_interpolation_arguments i18n.t(key, locale)
+        if value
+          ["#{locale}.#{key}", value]
         end
-        extract_interpolation_arguments i18n.t(key, locale)
-      end.compact
+      end.compact.to_h
+
+      next if interpolation_arguments.blank?
+      next if interpolation_arguments.values.uniq.length == 1
+      if ALLOWED_INTERPOLATION_MISMATCH_KEYS.include?(key)
+        missing_interpolation_argument_keys.push(key)
+        next
+      end
 
-      missing_interpolation_argument_keys.push(key) if interpolation_arguments.uniq.length > 1
+      # interpolation_arguments is a hash where the keys are the locale-specific content key,
+      # and values are the Set of interpolation arguments used in that key.
+      #
+      # We group and sort by the Set of interpolation arguments and assume the group with the
+      # most common interpolation arguments is the correct one. We then take the keys
+      # in the remaining groups and add them to the missing keys list.
+      keys = interpolation_arguments.group_by { |k, v| v }.sort_by { |k, v| v.length * -1 }.drop(1).
```

Comment:
surprised that we don't get unused argument failures?
```suggestion
      keys = interpolation_arguments.group_by { |k, _v| v }.sort_by { |_k, v| v.length * -1 }.drop(1).
```

Reply:
hopefully fixed here: https://github.com/18F/identity-idp/pull/10787